multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/api/impl/FilesApiServiceImpl.java

```diff
@@ -233,20 +268,29 @@ private AsyncUploadJobEntry getExistingJob(String spaceGuid, String namespace, S
                                    .user(SecurityContextUtil.getUsername())
                                    .namespace(namespace)
                                    .url(url)
+                                   .instanceIndex(configuration.getApplicationInstanceIndex())
```

Comment:
Will trigger a new upload, we're shutting the old deploy-service, so a new job should be triggered

Reply:
I think the request will fail because of the x-cf-app-instance, since it points to the application that we're stopping which will retry the upload. When both run it will call the right one

Reply:
Yes, tested it with simulated bg-deploy of the deploy-service. 
When 2 apps are running it still sends requests to the one that the upload started because of x-cf-app-instance
When the old one is stopped the router returns 400 which retries the whole upload and succeeds, 400 is returned again because of x-cf-app-instance